exercises/concept/monster-attack/.meta/design.md

```diff
@@ -0,0 +1,29 @@
+# Design
+
+## Learning objectives
+
+- Know that calling a function without all its arguments returns a new function.
+- Understand the arrows notation `->` in type annotations.
+- Know how convenient is partial applications in pipes `|>`.
+- Understand how partial application and pipes encourage placement of the main data structure as last argument of functions.
+
+## Out of scope
+
+- Composition of partial functions is out of scope, left for another concept.
+
+## Concepts
+
+- partial-application-composition
+
+## Prerequisites
+
+- basics-2
+
+## Analyzer
+
+This exercise could benefit from the following rules added to the the [analyzer][analyzer]:
+
+- Verify that the pipeline operator is used in `attack`
+- Verify that attackWithClaw and attahWithSword are only used once each in `attack`
```

Comment:
The idea is to use a partially applied function, like `annalynAttack = attackWithSword2 5`. You can take a look at the exemplar to see what I mean.

I think this would help a student learn partial application / currying, and it probably improves readability, although it is less concise. It's another example when currying can be useful. We could maybe add the name of the attacker (Annalyn / Kazak) in to the attack functions to make this more impactful, but it adds complexity as well, so not sure it is worth it.